tests/tests_full_integration_dyndns.yml

```diff
@@ -0,0 +1,186 @@
+      win_command: dnscmd.exe /config {{ network_ad }} /allowupdate 1
+    - name: Disable dns forwarders
+      win_command: dnscmd.exe /config /norecursion 1
+
+- name: Ensure that the role configures dynamic dns
+  hosts: client
+  tasks:
+    - name: Run the integration
+      when: "'ad' in groups and groups['ad']"
+      block:
+        - name: Debug dyndns settings
+          debug:
+            msg: >-
+              Interface {{ _ad_dyndns_iface | default('eth0') }} and server
+              : {{ hostvars[groups['ad'][0]].ansible_host }}
+        - name: Run the system role with proper config
+          include_role:
+            name: linux-system-roles.ad_integration
+          vars:
+            ad_dyndns_server: "{{ hostvars[groups['ad'][0]].ansible_host }}"
+            ad_dyndns_iface: "{{ _ad_dyndns_iface | default('eth0') }}"
+            ad_dyndns_auth: "none"
+            ad_dyndns_update: true
+            ad_dyndns_refresh_interval: 60
+        - name: Apply additional changes on sssd
+          ini_file:
+            path: /etc/sssd/sssd.conf
+            state: present
+            section: "domain/{{ ad_integration_realm | lower }}"
+            option: "{{ item.key }}"
+            value: "{{ item.value }}"
+            create: true
+            owner: root
+            group: root
+            mode: 0600
+          loop:
+            - key: debug_level
+              value: 9
+        - name: Clean sssd log
+          command: >-
+            truncate -s 0  /var/log/sssd/sssd_{{ ad_integration_realm }}.log
+        - name: Restart sssd
+          service:
+            name: sssd
+            status: restarted
+        - name: Pause for 5 to give sssd chance to refresh dn record on AD
+          ansible.builtin.pause:
+            minutes: 5
+        - name: Check sssd log fo dydndns update info
+          command: >-
+            grep -A 20 "nsupdate"  /var/log/sssd/sssd_{{ ad_integration_realm }}.log
+        - name: Grab sssd.conf contents
+          command: cat /etc/sssd/sssd.conf
+          register: command_output
+        - name: Print sssd.conf
+          debug:
+            msg: "{{ command_output.stdout }}"
+        - name: Get IP for host's FQDN
+          command: >-
+            dig +short
+            {{ ansible_fqdn }}.{{ hostvars[groups['client'][0]].ad_integration_realm }} A
+          register: dig_hostname
+          changed_when: false
+          failed_when: false
+        - name: Get hostname for host's IP address
+          command: "dig +short -x {{ ansible_default_ipv4.address }} PTR"
+          register: dig_ip
+          changed_when: false
+          failed_when: false
+        - name: Print to console dig outputs
+          debug:
+            msg:
+              - "Dig hostname: {{ dig_hostname.stdout }}"
+              - "Dig ip: {{ dig_ip.stdout }}"
+
+- name: Check hosts on AD
+  hosts: ad
+  gather_facts: true
+  tasks:
+    - name: AD checks
+      block:
```

Comment:
I don't think you need the `block` here - a `block` is usually used when
* you need to use `rescue` and/or `always`
* you want to define some variables that have block scope, to be used for several tasks
* you have a `when` condition that applies to several tasks

Reply:
I was planning to use always block with realm leave (delegate_to: client) but for some reason it just did not work.